Issue #41, add Network Element Overload Handling section (#56)
ietf-wg-scone/appman: draft-ietf-scone-applicability-manageability.md

element. Support for SCONE indication and bit-rate self-adaptation reduces complexity and CPU processing
load in the network element.

## Network Element Overload Handling

Processing SCONE packets creates a per-flow update obligation for
network elements. To avoid throughput advice expiring
({{Section 5.4 of I-D.ietf-scone-protocol}}), each active flow
needs at least one rate signal update within each monitoring
period. When the number of concurrent SCONE flows grows unusually
large, as can occur during periods of elevated traffic or network

> stress, this update obligation can become a processing concern.

Reviewer: Why does the number of flows grow during periods of network stress? I think the number of scone packets after all is still very small compared to generally packet processing.

Reviewer 2: @mirjak Think attacks. Attackers can easily send millions of SCONE packets, maybe each with different address/ports.

Author: based on Christian's comment, no change needed. Other suggestions are accepted. will merge the PR.

Network element vendors and operators have two broad options for
managing this. One is to provision the network element with
capacity that accommodates the expected range of concurrent flows,
recognizing that what constitutes adequate capacity will vary by
deployment. The other is to handle the condition gracefully when
that capacity is reached. One approach for graceful handling is
for the network element to stop updating SCONE packets for a
subset of flows entirely, rather than attempting to serve all

> flows with intermittent or delayed updates. A flow that goes

Reviewer 3: / suggest: /A flow that does not observe a SCONE update/

> flows with intermittent or delayed updates. A flow that goes

Author: ```suggestion
flows with intermittent or delayed updates. A flow that does not observe a SCONE update
```

> without a SCONE update for a full monitoring period will have its

Author: ```suggestion
for a full monitoring period will have its
```

throughput advice expire, causing the endpoint to operate without

> SCONE-advised limits. This

Reviewer 3: /SCONE-advised limits/SCONE-advice/

> SCONE-advised limits. This

Author: ```suggestion
SCONE-advice. This
```

outcome is more predictable than partial updates, which can cause

> endpoints to alternate between operating under SCONE throughput

Author: ```suggestion
endpoints to alternate between operating with SCONE throughput
```

> endpoints to alternate between operating under SCONE throughput

Reviewer 3: /under/with/

advice and operating without it. Which flows continue to receive
updates and how that selection changes over time are operator
policy decisions. Network element implementations can provide the
configuration options through which operators define and adjust
that policy.

> 

Reviewer 4: There is one another option to handle this scenario. Operators/NE providers to identify legit flows by using existing mechanism such as SNI. This would avoid the need of NE to update the scone pkts for flows which are not legit.

Author: @anooptomar29, this approach depends on SNI being visible in the clear at the network element. With ECH, the SNI is encrypted, so the network element would have no legit/non-legit signal to filter on. So should this be scoped to non-ECH deployments, or is there a fallback you have in mind for when SNI is encrypted?

Also, would this sit as a third option alongside the two already in this section (provision for peak / graceful shedding), or as a prerequisite step before either applies?

Reviewer 4: @smishra1200 This can be another option for CSPs to address this concern of processing need for large number of  flows generated by attackers.  Regarding the 1st point about ECH making SNI private key encrypted making it unreadable at NE. With ECH, generic SNI can still be sent in unencrypted Outer ClientHello in clear text and can be read by NE to address this concern. Essentially CSPs can agree with legit app providers on the usage of known SNI and NE can process scone pkt on flows ( on which known SNI was sent during ICH).

Reviewer 4: @smishra1200 I dont see my comment/suggestion to add another option for CSPs to address this concern of processing load in the PR changes. I have created https://github.com/ietf-wg-scone/appman/pull/90 to make sure this is not lost.

network elements. To avoid throughput advice expiring
({{Section 5.4 of I-D.ietf-scone-protocol}}), each active flow
needs at least one rate signal update within each monitoring
period. When the number of concurrent SCONE flows grows unusually
large, as can occur during periods of elevated traffic or network
stress, this update obligation can become a processing concern.

Network element vendors and operators have two broad options for
managing this. One is to provision the network element with
capacity that accommodates the expected range of concurrent flows,
recognizing that what constitutes adequate capacity will vary by
deployment. The other is to handle the condition gracefully when
that capacity is reached. One approach for graceful handling is
for the network element to stop updating SCONE packets for a
subset of flows entirely, rather than attempting to serve all
flows with intermittent or delayed updates. A flow that goes
without a SCONE update for a full monitoring period will have its
throughput advice expire, causing the endpoint to operate without
SCONE-advised limits, relying instead on standard congestion

> control ({{Section 5.4 of I-D.ietf-scone-protocol}}). This

Reviewer 5: ```suggestion
SCONE-advised limits. This
```

While this is correct, I'm a little worried about the phrasing here. Flows always rely on standard congestion control, it's just that they might become application limited on SCONE signals. 
I think the sentence can end before mentioning congestion control.
